File: Kernel Simulator/Shell/Commands/ShowNotifs.vb

'    Kernel Simulator  Copyright (C) 2018-2022  EoflaOE
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Imports KS.Misc.Notifications

Namespace Shell.Commands
    Class ShowNotifsCommand
        Inherits CommandExecutor
        Implements ICommand

        Public Overrides Sub Execute(StringArgs As String, ListArgs() As String, ListArgsOnly As String(), ListSwitchesOnly As String()) Implements ICommand.Execute
            Dim Count As Integer = 1
            If Not NotifRecents.Count = 0 Then
                For Each Notif As Notification In NotifRecents
                    TextWriterColor.Write($"[{Count}/{NotifRecents.Count}] {Notif.Title}: ", False, ColTypes.ListEntry)
                    TextWriterColor.Write(Notif.Desc, False, ColTypes.ListValue)
                    If Notif.Type = NotifType.Progress Then
                        TextWriterColor.Write($" ({Notif.Progress}%)", False, If(Notif.ProgressFailed, ColTypes.Error, ColTypes.Success))
                    End If
                    TextWriterColor.Write("", True, ColTypes.Neutral)
                    Count += 1
                Next
            Else
                TextWriterColor.Write(DoTranslation("No recent notifications"), True, ColTypes.Neutral)
            End If
        End Sub

    End Class
End Namespace

File: Kernel Simulator/Shell/Commands/ShowNotifs.cs
﻿//
// Kernel Simulator  Copyright (C) 2018-2024  Aptivi
//
// This file is part of Kernel Simulator
//
// Kernel Simulator is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// Kernel Simulator is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY, without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

using KS.ConsoleBase.Colors;
using KS.Languages;
using KS.Misc.Notifiers;
using KS.ConsoleBase.Writers;
using KS.Shell.ShellBase.Commands;

namespace KS.Shell.Commands
{
    class ShowNotifsCommand : CommandExecutor, ICommand
    {

        public override void Execute(string StringArgs, string[] ListArgs, string[] ListArgsOnly, string[] ListSwitchesOnly)
        {
            int Count = 1;
            if (!(Notifications.NotifRecents.Count == 0))
            {
                foreach (Notification Notif in Notifications.NotifRecents)
                {
                    TextWriters.Write($"[{Count}/{Notifications.NotifRecents.Count}] {Notif.Title}: ", false, KernelColorTools.ColTypes.ListEntry);
                    TextWriters.Write(Notif.Desc, false, KernelColorTools.ColTypes.ListValue);
                    if (Notif.Type == Notifications.NotifType.Progress)
                    {
                        TextWriters.Write($" ({Notif.Progress}%)", false, Notif.ProgressFailed ? KernelColorTools.ColTypes.Error : KernelColorTools.ColTypes.Success);
                    }
                    TextWriters.Write("", true, KernelColorTools.ColTypes.Neutral);
                    Count += 1;
                }
            }
            else
            {
                TextWriters.Write(Translate.DoTranslation("No recent notifications"), true, KernelColorTools.ColTypes.Neutral);
            }
        }

    }
}